.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,61 @@
+name: Publish Packages to NPM
+
+on:
+  push:
+    branches:
+      - 'publishable_package'
+
+jobs:
+  affected-packages:
+    name: List affected publishable packages
+    runs-on: ubuntu-latest
+    outputs:
+      affected: ${{ steps.set-affected.outputs.affected }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v3
```

Comment:
fetch depth 0 + nrwl/nx-set-shas@v3 are required for using affected. FYI for the future